docs/guide/declarative_containers.md

```diff
@@ -0,0 +1,111 @@
+# Declarative Environment Configuration
+
+jobq allows you to define your Docker environment in a declarative YAML format. Instead of writing a Dockerfile directly.
+
+## YAML Structure
+
+The YAML has the following structure.
+```yaml
+build:
+  base_image: <base_image>
+  dependencies:
+    apt: [<apt_packages>]
+    pip: [<pip_packages>]
+  volumes:
+    - <host_path>:<container_path>
+  user:
+    name: <username>
+  config:
+    env:
+      - <env_var>: <value>
+    arg:
+      - <build_arg>: <value>
+    stopsignal: <signal>
+    shell: <shell>
+  meta:
+    labels:
+      - <label_key>: <label_value>
+  workdir: <working_directory>
+  filesystem:
+    copy:
+      - <src>: <dest>
+    add:
+      - <src>: <dest>
+```
+Let us walk through an example of each of the options.
+
+The YAML file uses a `build` key as the root, under which various aspects of the Docker image are defined.
+
+First, you have to first specify the base image of the Dockerfile, for example:
+
+```yaml
+build:
+  base_image: python:3.12-slim
+```
+
+Then, you can define system level and Python dependencies.
+
+```yaml
+build:
+  dependencies:
+    apt: [curl, git]
+    pip: [attrs, pyyaml, test.whl, marker-package, -e.]
+```
+In order to make external storage available, you can define volume mapping. In this case, the build directory.
```

Comment:
Here, we should mention the build context resolution, since relative paths are considered relative to the build context, not the image spec file (IIRC, please double check).